app/celery/v3/notification_tasks.py

```diff
@@ -269,13 +285,35 @@ def v3_send_sms_notification(notification: Notification, sender_phone_number: st
     return
 
 
-def v3_persist_failed_notification(notification: Notification, status: str, status_reason: str):
+def v3_persist_permanent_failure(notification: Notification):
+    try:
+        notification_json = notification.serialize_permanent_failure()
+        notification_failure = NotificationFailures(
+            notification_id=notification.id,
+            body=notification_json
+        )
+        db.session.add(notification_failure)
+        db.session.commit()
+    except Exception as err:
+        db.session.rollback()
+        current_app.logger.error("Unable to save permanent failure. Error: '%s'", err)
```

Comment:
I would make this "critical" because it indicates a loss of information.

Reply:
done